src/cpp/common/alarm.cc

```diff
@@ -31,21 +31,17 @@
  */
 
 #include <grpc++/alarm.h>
-#include <grpc++/completion_queue.h>
 #include <grpc++/impl/grpc_library.h>
-#include <grpc/grpc.h>
 
 namespace grpc {
 
 static internal::GrpcLibraryInitializer g_gli_initializer;
-Alarm::Alarm(CompletionQueue* cq, gpr_timespec deadline, void* tag)
-    : tag_(tag),
-      alarm_(grpc_alarm_create(cq->cq(), deadline, static_cast<void*>(&tag_))) {
+
+Alarm::~Alarm() {
   g_gli_initializer.summon();
```

Comment:
Why did this g_gli_initializer move from the constructor to the destructor? As a side note, is it even possible that this is relevant, given that CompletionQueue is one of the arguments?